src/typeInference/annotateUnaryOperator.ts

```diff
@@ -0,0 +1,64 @@
+import isSubtypeOf from '../expressions/dataTypes/isSubtypeOf';
+import { SequenceType, ValueType } from '../expressions/dataTypes/Value';
+import astHelper, { IAST } from '../parsing/astHelper';
+
+/**
+ * Adds the unary minus operator type annotation to the AST
+ * The unary minus returns the value of its operand with the sign negated (-2 = -2 and --2 = 2).
+ *
+ * @param ast A reference to the AST, so a new node can be inserted.
+ * @param valueType The type of the value that the operator is called on.
+ * @returns An appropriate SequenceType if the operation was valid, undefined if not.
+ */
+export function annotateUnaryMinus(
+	ast: IAST,
+	valueType: SequenceType | undefined
+): SequenceType | undefined {
+	// If we don't now the child type, we can't infer the current type
+	if (!valueType) {
+		return undefined;
```

Comment:
I think you can: it's always a numeric type. It's just that we do not know what numeric typ

Reply:
Yeah returning a XSNUMERIC is a good idea. This has been fixed in our current dev branch

Reply:
Can you update the PR with that branch somewhere this week? that way this comment will resolve itself